mitodl/mitxonline
# Implemented country blocklist at the course level

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #163 
Closes #105 

#### What's this PR do?
- Adds `django-countries` as a dependency
- Adds a model for keeping track of countries that are blocked from enrollment for a given course
- Shows an alert to users who attempt to enroll in a course if their country is blocked
- Fixes some Django admin issues in the courses app

#### How should this be manually tested?
- Pick a user and note their country (`user.legal_address.country`)
- Add a `BlockedCountry` for that user's country and the parent course of a course run you want to enroll in
- Visit the detail page of that course run and click "Enroll Now"

You should see an error message indicating that your country is blocked from enrollment. If you remove that `BlockedCountry` record, it should succeed

#### Screenshots (if appropriate)
![ss 2021-09-10 at 24 06 17 ](https://user-images.githubusercontent.com/14932219/132802525-b1bfea5a-b1f7-40e4-9ea2-915062617e6e.png)
